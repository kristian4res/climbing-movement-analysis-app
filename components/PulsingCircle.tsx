import { PulsingCircleProps } from '@/interfaces';
import React, { useEffect } from 'react';
import Animated, { Easing, useAnimatedStyle, useSharedValue, withRepeat, withTiming } from 'react-native-reanimated';

const PulsingCircle: React.FC<PulsingCircleProps> = ({ isActive }) => {
    const scale = useSharedValue(1);

    useEffect(() => {
        if (isActive) {
            scale.value = withRepeat(
            withTiming(1.3, {
                duration: 500,
                easing: Easing.ease,
            }),
            -1,
            true
            );
        } else {
            scale.value = 1;
        }
    }, [isActive]);

    const animatedStyle = useAnimatedStyle(() => {
        return {
            transform: [{ scale: scale.value }],
        };
    });

    return (
        <Animated.View
            testID={'pulsing-circle'}
            style={[
            {
                position: 'absolute',
                zIndex: 11,
                top: 20,
                width: 20,
                height: 20,
                borderRadius: 10,
                backgroundColor: isActive ? 'red' : 'white',
            },
            animatedStyle,
            ]}
        />
    );
};

export default PulsingCircle;