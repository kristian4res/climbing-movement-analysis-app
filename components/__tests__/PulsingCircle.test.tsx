import React from 'react';
import { render } from '@testing-library/react-native';
import PulsingCircle from '@/components/PulsingCircle';

describe('PulsingCircle component', () => {
    it('matches the snapshot', () => {
        const tree = render(<PulsingCircle isActive={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with red background when active', () => {
        const { getByTestId } = render(<PulsingCircle isActive={true} />);
        expect(getByTestId('pulsing-circle').props.style).toEqual({backgroundColor: "red", borderRadius: 10, height: 20, position: "absolute", top: 20, transform: [{scale: 1}], width: 20, zIndex: 11});
    });

    it('renders with white background when inactive', () => {
        const { getByTestId } = render(<PulsingCircle isActive={false} />);
        expect(getByTestId('pulsing-circle').props.style).toEqual({backgroundColor: "white", borderRadius: 10, height: 20, position: "absolute", top: 20, transform: [{scale: 1}], width: 20, zIndex: 11});
    });
});