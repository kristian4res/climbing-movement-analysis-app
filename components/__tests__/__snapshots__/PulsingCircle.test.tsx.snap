// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulsingCircle component matches the snapshot 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
    }
  }
  collapsable={false}
  style={
    {
      "backgroundColor": "red",
      "borderRadius": 10,
      "height": 20,
      "position": "absolute",
      "top": 20,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": 20,
      "zIndex": 11,
    }
  }
  testID="pulsing-circle"
/>
`;
